apm-sniffer/apm-sdk-plugin/mysql-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v5/ConnectionCache.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.jdbc.mysql.v5;
+
+import org.apache.skywalking.apm.plugin.jdbc.trace.ConnectionInfo;
+import org.apache.skywalking.apm.util.StringUtil;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author: dingshaocheng
+ */
+public class ConnectionCache {
+    private static Map<String, ConnectionInfo> CONNECTIONS_MAP = new ConcurrentHashMap<String, ConnectionInfo>();
+
+    public static ConnectionInfo get(String host, String port) {
+        final String connStr = String.format("%s:%s", host, port);
+        return CONNECTIONS_MAP.get(connStr);
+    }
+
+    public static void save(ConnectionInfo connectionInfo) {
+        Arrays.stream(connectionInfo.getDatabasePeer()
```

Comment:
I'm afraid that it can't work in JDK 1.6, because the agent needs to support JDK 1.6

Reply:
All agent plugin should be compiled in JDK6, if setting is right. @ascrutae

Reply:
From the CI result, I think this plugin pom setting has some issue. @dsc6636926 Could you recheck?

Reply:
@wu-sheng The `Arrays.stream` API is from JDK 1.8, and I think it's can't run in JDK 1.8

Reply:
You don't read my words clearly. I mean, if this codes are not right, but CI passed, then there is another issue.